### DISCUSSION TITLE
Bitcoin Summer of code

### DISCUSSION THREAD

**Discussion by sanket1729** [rust-bitcoin/rust-bitcoin#832] [Ideas]
Bitcoin Summer of code(https://www.summerofbitcoin.org/) is around the corner. Think of it like GSoC, but for bitcoin. We will have several students who are new to both rust and bitcoin looking at the repository to contribute. 

If there are any contributors who are willing to mentor any new project, do let me know and I can forward your proposal. For example: See https://docs.google.com/document/d/1xedUdFmw-YNu0VBstgQSWS9piXMGGZobSTEcp7wPTw0/edit. (Sorry for the google link :(, if it's inconvenient to access, I can create a gist). Roughly you would be required to meet once per week for an hour with your mentee. The program would roughly be 2.5 months. It would be a good opportunity to mentor new bitcoiners while getting some open source work done!

Would it be worth creating a new tag for "bitcoin summer of code" to guide students to new issues? Maybe it is the same as Good first issue. With this, we can create new issues with a template like 

Summer of Bitcoin issue:

- Required bitcoin reading: 
...
- Required bitcoin/rust skills:
...
- Expected outcome:
...
- Diifficulty :
...
This is just starters, any new ideas are welcome.

**Comment by tcharding** [comment]
I'm willing to mentor someone. Here is a project description: https://gist.github.com/tcharding/316b0c2e121e333d7496d89c761a726a

**Comment by sanket1729** [comment]
@tcharding awesome, I will share it with the organizers.

**Comment by sanket1729** [comment]
HI @tcharding , the project is added at https://www.summerofbitcoin.org/project-ideas-details?recordId=rec6X8GnL7bzPHg8K. Looks like the organizers clubbed it with rust-miniscript because I requested it. I have asked the title to be updated to rust-bitcoin. 
You are also added on the mentors page here: https://www.summerofbitcoin.org/mentors

**Comment by sanket1729** [comment]
@tcharding, I would also suggest creating multiple "Good first issues" as many new students are looking for hands-on things to get started.
